packages/alfa-dom/src/node.ts

```diff
@@ -260,6 +262,30 @@ export abstract class Node
 
     return true;
   }
+
+  /**
+   * Nodes that represents nothing and thus should not be rendered nor included in the accessibility tree.
+   * These are `<template>` elements and descendants of `<iframe>`.
+   *
+   * It is not fully clear what else should fit here. Notably because other non-void elements with "Nothing" as
+   * content model (option and colgroup, in some cases) tend to be weirdly patched up by HTML parsing (child
+   * causes an early termination of the element and is pushed out of it).
+   *
+   * We may also want to add other content that is only used as fallback, such as some children of `<object>`
+   * or `<video>`. These seem to be elements that are in a "transparent" content model…
+   */
+  public representsNothing(): boolean {
+    return (
+      // @see https://html.spec.whatwg.org/multipage/scripting.html#the-template-element
+      // "In a rendering, the template element represents nothing."
+      and(Element.isElement, hasName("template"))(this) ||
+
+      // @see https://html.spec.whatwg.org/multipage/iframe-embed-object.html#the-iframe-element
+      // "Descendants of iframe elements represent nothing. (In legacy user agents that do not support iframe elements,
+      // the contents would be parsed as markup that could act as fallback content.)"
+      this.ancestors().some(and(Element.isElement, hasName("iframe")))
+    )
+  }
```

Comment:
I'm not too keen on introducing this API as it doesn't have a well-defined purpose other than catering to our immediate needs in the construction of the accessibility tree. As you note in the docblock, it's not that clear what the API _should do_.

I'd dig around Servo and Chromium for hints on how they handle this. If that doesn't provide any useful insight, just inline the needed bits so we don't introduce public API with no clear purpose.

Reply:
Yes, I'm also far from happy to put that here…
The point was that it is used both in building the accessibility tree, and in `isRendering`. So I wasn't sure where to put it in order to share it…

Reply:
Only the bit about `<iframe>` descendants is needed for those though; `<template>` elements are hidden by the UA style sheet:

https://github.com/Siteimprove/alfa/blob/7ad79c9899dc836df9d2b305ea27ba6ef8225141/packages/alfa-cascade/src/user-agent.ts#L14-L19

All that's left is then just a single line of code which is easily inlined.

Reply:
🤔 but then shouldn't we rather just update that stylesheet to hide children of `<iframe>`?

Reply:
That's one option, and probably the easiest solution for lack of a layout system 😄 I just dug through Chromium and it ignores all nodes without an associated layout object when building its accessibility tree:

https://github.com/chromium/chromium/blob/f9a52600a8d706ca524e8b10b2fc0fbbde992b57/third_party/blink/renderer/modules/accessibility/ax_layout_object.cc#L627-L631

Reply:
Tried to hide children of `<iframe>` in UA style sheet, but my CSS-fu is not good enough 😖 
https://github.com/Siteimprove/alfa/pull/383/files#diff-5be8aa0a1cd557655c16d24015883e65R229-R234

I'm not happy with the current solution that duplicates the logic, especially the but in `isRendered` is in a unlikely place and prone to be forgotten later 😢 

Do you have a better idea for handling this?

Reply:
Until we have a layout system in place then any solution that doesn't introduce additional public API surface is a good solution to me.

Reply:
👍 Then let's do it that way.